src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c

```diff
@@ -28,12 +29,48 @@ const ActorInit Obj_Blockstop_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void ObjBlockstop_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjBlockstop* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/ObjBlockstop_Init.s")
+    if (Flags_GetSwitch(globalCtx, (s32)this->actor.params)) {
```

Comment:
```suggestion
    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
```
Arithmetic casts in functions are almost always not needed.